src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -161,5 +180,136 @@ public Task ExecuteChildContentAsync()
 
             return _childContent;
         }
+
+        private class ItemsCopyOnWriteDictionary : IDictionary<string, object>
```

Comment:
Can't do this sadly, `TagHelperExecutionContext` is the scope manager for executing `TagHelper`s, therefore if this existed in the `TagHelperContext` then we'd be wrapping the `TagHelperExecutionContext`s `Items` which would then never be updated and never passed down to children.